app/code/Magento/Tax/Model/Calculation/AbstractCalculator.php

```diff
@@ -408,14 +410,14 @@ protected function deltaRound($price, $rate, $direction, $type = self::KEY_REGUL
         if ($price) {
             $rate = (string)$rate;
             $type = $type . $direction;
-            // initialize the delta to a small number to avoid non-deterministic behavior with rounding of 0.5
-            $delta = isset($this->roundingDeltas[$type][$rate]) ?
-                $this->roundingDeltas[$type][$rate] :
-                0.000001;
-            $price += $delta;
             $roundPrice = $price;
             if ($round) {
```

Comment:
I believe this may be simplified even further by making

```
if (!$round) {
    return $price;
}
```

in the very beginning.

Reply:
Thanks @orlangur, I will update with suggested changes.